Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/Microsoft_Defender_For_Endpoint_-_Collect_investigation_package_.yml

```diff
@@ -0,0 +1,440 @@
+            value: {}
+    view: |-
+      {
+        "position": {
+          "x": 162.5,
+          "y": 370
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "8":
+    id: "8"
+    taskid: 59afffe4-dd31-45ee-8111-af067040c394
+    type: title
+    task:
+      id: 59afffe4-dd31-45ee-8111-af067040c394
+      version: -1
+      name: Done
+      type: title
+      iscommand: false
+      brand: ""
+      description: ''
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 162.5,
+          "y": 1560
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "9":
+    id: "9"
+    taskid: 2fb5923c-1661-4cd2-81ae-2800bc471c6a
+    type: playbook
+    task:
+      id: 2fb5923c-1661-4cd2-81ae-2800bc471c6a
+      version: -1
+      name: GenericPolling
+      description: |-
+        Use this playbook as a sub-playbook to block execution of the master playbook until a remote action is complete.
+        This playbook implements polling by continuously running the command in Step \#2 until the operation completes.
+        The remote action should have the following structure:
+
+        1. Initiate the operation.
+        2. Poll to check if the operation completed.
+        3. (optional) Get the results of the operation.
+      playbookName: GenericPolling
+      type: playbook
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "5"
+    scriptarguments:
+      Ids:
+        simple: ${MicrosoftATP.MachineAction.ID}
+      Interval:
+        simple: "1"
+      PollingCommandArgName:
+        simple: id
+      PollingCommandName:
+        simple: microsoft-atp-list-machine-actions-details
+      Timeout:
+        simple: "10"
+      dt:
+        simple: MicrosoftATP.MachineAction(val.Status != 'Succeeded').ID
+    separatecontext: true
+    loop:
+      iscommand: false
+      exitCondition: ""
+      wait: 1
+      max: 100
+    view: |-
+      {
+        "position": {
+          "x": 390,
+          "y": 1070
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "10":
+    id: "10"
+    taskid: fcda1408-bf90-48f8-8018-d9e4449186f6
+    type: regular
+    task:
+      id: fcda1408-bf90-48f8-8018-d9e4449186f6
+      version: -1
+      name: Download File
+      description: Sends http request. Returns the response as json.
+      scriptName: http
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "8"
+    scriptarguments:
+      filename:
+        simple: Investigation_package_incident_${incident.id}.zip
```

Comment:
You can add a task to unzip

Reply:
I thought about that - but inside the zip itself, you have multiple folders and another bunch of files that probably will create a mess inside the context (instead of actually providing good info). That's why I left it as it is.